data_structures/queue_using_two_stacks.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @author [shoniavika](https://github.com/shoniavika)
+ * @file
+ *
+ * Implementation of a Queue using two Stacks.
+ */
+
+#include <cassert>
+#include <iostream>
+
+#include "./stack.h"
+
+/**
+ * Queue data structure. Stores elements in FIFO
+ * (first-in-first-out) manner.
+ *     methods:
+ *         push(int x)  :  Pushes x to the back of queue.
+ *         pop()        :  Removes an element from the front
+ *         peek()       :  Returns first element, without removing
+ *         empty()      :  Returns whether the queue is empty
+ */
+class MyQueue {
+ private:
+    stack<int> s1, s2;
+
+    /**
+     * Helper function for rearranging elements in stacks
+     */
+    void rearrange();
+
+ public:
+    /**
+     * Constructor for queue.
+     */
+    MyQueue() = default;
+
+    /**
+     * Pushes x to the back of queue.
+     */
+    void push(int x);
+
+    /**
+     * Removes an element from the front of the queue.
+     */
+    int pop();
+
+    /**
+     * Returns first element, without removing it.
+     */
+    int peek();
+
+    /**
+     * Returns whether the queue is empty.
+     */
+    bool empty();
+};
+
+/**
+ * Appends element to the end of the queue
+ */
+void MyQueue::push(int x) { s1.push(x); }
+
+/**
+ * Removes element from the front of the queue
+ */
+int MyQueue::pop() {
+    rearrange();
+    int temp = 0;
+    if (!empty()) {
+        temp = s2.top();
+        s2.pop();
+    } else {
+        std::cout << "Queue is empty" << std::endl;
+    }
+    return temp;
+}
+
+/**
+ * Helper function for rearranging elements from
+ * one stack into another.
+ */
+void MyQueue::rearrange() {
+    if (s2.isEmptyStack()) {
+        while (!s1.isEmptyStack()) {
+            int temp = s1.top();
+            s1.pop();
+            s2.push(temp);
+        }
+    }
+}
+
+/**
+ * Returns element in the front.
+ * Does not remove it.
+ */
+int MyQueue::peek() {
```

Comment:
```suggestion
template<typename T>
T MyQueue<T>::peek() {
```
Also, you can add the warning message as above